samtranslator/model/preferences/deployment_preference_collection.py

```diff
@@ -226,7 +226,13 @@ def _convert_alarms(self, preference_alarms):
             If Alarms is in the wrong format
         """
         if not preference_alarms or is_intrinsic_no_value(preference_alarms):
-            return None
+            # return the empty alarms configuration in case if there is no alarms at all defined in the
+            # DeploymentPreference to force CFN to update the deployment group and remove the alarms values from the
+            # created DeploymentGroup. Check issue https://github.com/aws/serverless-application-model/issues/2452
+            return {
```

Comment:
Is there any concern that it would be a breaking change? Or put it other way - is there any concern that there could be some customers relying on this incorrect behaviour?

Reply:
The only thing that I can think about, if the customers did not remove the actual alarm, but removed the alarm property from the deployment preference. The customer will find that some of the deployments will be rolled back because of this alarm, and they want this behaviour. But I do not think that this is a real scenario, I always expect that when the customers remove the alarm configuration, they should expect no rollbacks